### PR TITLE
added source to range - without breaking pre-existing topic

### PR DESCRIPTION
**Pull request by supertick**

### CODE REVIEW

**Comment by supertick** [comment]
greg
added source to range - without breaking pre-existing topic

File: src/main/java/org/myrobotlab/service/UltrasonicSensor.java [hunk 1]
```diff
@@ -74,7 +75,8 @@ public UltrasonicSensor(String n, String id) {
   // change units, sample rate, etc
   // FIXME - NOT SERVICE .. possibly name or interface but not service
   public void addRangeListener(Service service) {
-    addListener("publishRange", service.getName(), "onRange");
+    addListener(""
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/UltrasonicSensor.java]: What is "ge"  ?

File: src/main/java/org/myrobotlab/service/data/AbstractData.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * An attempt to try to normalize some of the common fields of these pojos
+ * 
+ * @author grog
+ *
+ */
+public abstract class AbstractData {
+
+  public String source;
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/data/AbstractData.java:11]: It's a bit overkill to create an abstract base class for this...  don't you think?

File: src/main/java/org/myrobotlab/service/UltrasonicSensor.java (hunk 1 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/UltrasonicSensor.java]: what the hell ? ... pre-coffee mess ... thanks for the catch

File: src/main/java/org/myrobotlab/service/data/AbstractData.java (hunk 2 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/data/AbstractData.java:11]: I don't think so - as soon as anyone gets more than one of the same type of sensor - regardless of the type, they will want to know the origin of the sensor data, and source/name can sort it out

**Pull request merged**